core/src/main/java/tc/oc/pgm/namedecorations/NameDecorationProvider.java

```diff
@@ -1,11 +1,16 @@
 package tc.oc.pgm.namedecorations;
 
 import java.util.UUID;
+import net.kyori.text.Component;
 
 /** This interface is intended to return what prefix and suffix a player should have */
 public interface NameDecorationProvider {
 
   String getPrefix(UUID uuid);
 
   String getSuffix(UUID uuid);
+
+  Component getPrefixComponent(UUID uuid);
+
+  Component getSuffixComponent(UUID uuid);
```

Comment:
```suggestion
  default Component getPrefixComponent(UUID uuid) {
      return TextComponent.of(getPrefix(uuid));
  }

  default Component getSuffixComponent(UUID uuid) {
      return TextComponent.of(getSuffix(uuid));
  }
```
Keeping this interface simple to implement for 3rd parties is relevant IMO, and it's an easy enough default. (Maybe do a component parsing from legacy text instead of just a plain text with the text)